src/services/CleanupService.ts

```diff
@@ -222,79 +205,154 @@ export const cleanCookies = async (
   });
 };
 
-/** This will use the browsingData's hostname attribute to delete any extra browsing data */
+// Cleanup of all cookies for domain.
+export const clearCookiesForThisDomain = async (
+  state: State,
+  tab: browser.tabs.Tab,
+) => {
+  const hostname = getHostname(tab.url);
+  const firstPartyIsolate = await isFirstPartyIsolate();
+  const getCookies = (await browser.cookies.getAll(
+    returnOptionalCookieAPIAttributes(state, {
+      domain: hostname,
+      storeId: tab.cookieStoreId,
+    }, firstPartyIsolate),
+  ));
+  const cookies = getCookies.filter(c => c.name !== LSCLEANUPNAME);
+
+  if (cookies.length > 0) {
+    let cookieDeletedCount = 0;
+    for (const cookie of cookies) {
+      const r = await browser.cookies.remove(returnOptionalCookieAPIAttributes(state, {
+        firstPartyDomain: cookie.firstPartyDomain,
+        name: cookie.name,
+        storeId: cookie.storeId,
+        url: prepareCookieDomain(cookie),
+      }, firstPartyIsolate) as {
+        // Fix type error with undefineds with cookies.remove
```

Comment:
Interesting, I wonder why is this?

Reply:
What are you referring to here?

Reply:
I'll keep this PR/branch until we clarify what is needed here.

Reply:
Why does `browser.cookies.remove` return undefined here? Is the return type not defined in the types library?